site/app/models/gradeable/GradeableUtils.php

```diff
@@ -116,4 +116,26 @@ public static function getGradeablesFromUserAndCourse(Core $core, array &$calend
             "graded_gradeables" => $graded_gradeables,
             "submit_btns" => $submit_btns];
     }
+
+    public static function vcsIntToStr(int $type): string {
+        return match ($type) {
+            0 => "vcs_type_submitty_unique",
+            1 => "vcs_type_submitty_shared",
+            2 => "vcs_type_external_public",
+            3 => "vcs_type_external_private_student_created",
+            4 => "vcs_type_external_private_instructor_created",
+            default => "vcs_none"
+        };
+    }
+
+    public static function vcsStrToInt(string $str): int {
+        return match ($str) {
+            "vcs_type_submitty_unique" => 0,
```

Comment:
This PR looks good so far even though its in draft, thanks for making this with the limited issue information.

Just as a heads up you might want to look through the code base to see if these strings can be replaced or if there are used values already, for example here a specific string to indicate which vcs type is used in configuration

https://github.com/Submitty/Submitty/pull/10188/files#diff-cea0d0636e0f8c01d014a281e408fcca63c771c6d7cb35087cdbda7615e35076R103